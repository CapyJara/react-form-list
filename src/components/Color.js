import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, hex, rgb }) {
  
  return (
    <dl>
      <dt>Name</dt>
      <dd>{name}</dd>

      <dt>Hex</dt>
      <dd>{hex}</dd>

      <dt>Rgb</dt>
      <dd>
        <p>Red: {rgb.red}</p>
        <p>Green: {rgb.green}</p>
        <p>Blue: {rgb.blue}</p>
      </dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.shape({
    red: PropTypes.number.isRequired,
    green: PropTypes.number.isRequired,
    blue: PropTypes.number.isRequired
  })
};

export default Color;
