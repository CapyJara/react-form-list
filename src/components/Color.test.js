import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('renders a Color', () => {
    const wrapper = shallow(<Color name={'pink'} hex={'bl234'}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
