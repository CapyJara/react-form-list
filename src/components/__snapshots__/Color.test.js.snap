// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component renders a Color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Red
  </dd>
  <dt>
    Hex
  </dt>
  <dd>
    bla
  </dd>
  <dt>
    Rgb
  </dt>
  <dd>
    <p>
      Red: 
      255
    </p>
    <p>
      Green: 
      0
    </p>
    <p>
      Blue: 
      0
    </p>
  </dd>
</dl>
`;
